simd/src/main/java/com/dylibso/chicory/simd/SimdInterpreterMachine.java

```diff
@@ -0,0 +1,399 @@
+        // this should be almost an in-place replace on the stack, should we measure performance
+        // against push and pop?
+
+        var offset = stack.size() - 2;
+        var not = LongVector.fromArray(LongVector.SPECIES_128, stack.array(), offset).not();
+        var res = not.toArray();
+
+        System.arraycopy(res, 0, stack.array(), offset, 2);
+    }
+
+    private static void V128_BITSELECT(MStack stack) {
+        // https://github.com/tetratelabs/wazero/blob/58488880a334e8bda5be0d715a24d8ddeb34c725/internal/engine/interpreter/interpreter.go#L2378C7-L2384
+        var cHi = stack.pop();
+        var cLo = stack.pop();
+        var x2Hi = stack.pop();
+        var x2Lo = stack.pop();
+        var x1Hi = stack.pop();
+        var x1Lo = stack.pop();
+        // v128.or(v128.and(v1, c), v128.and(v2, v128.not(c)))
+        stack.push((x1Lo & cLo) | (x2Lo & (cLo ^ 0xFFFFFFFFFFFFFFFFL)));
+        stack.push((x1Hi & cHi) | (x2Hi & (cHi ^ 0xFFFFFFFFFFFFFFFFL)));
+    }
+
+    private static void F32x4_MUL(MStack stack) {
+        var val1High = stack.pop();
+        var val1Low = stack.pop();
+        var val2High = stack.pop();
+        var val2Low = stack.pop();
+
+        long resultLow = 0L;
+        long resultHigh = 0L;
+
+        for (int i = 0; i < 2; i++) {
+            var shift = i * 32L;
+            resultHigh |=
+                    (Float.floatToIntBits(
+                                            Float.intBitsToFloat(
+                                                            (int)
+                                                                    ((val1High >> shift)
+                                                                            & 0xFFFFFFFFL))
+                                                    * Float.intBitsToFloat(
+                                                            (int)
+                                                                    ((val2High >> shift)
+                                                                            & 0xFFFFFFFFL)))
+                                    & 0xFFFFFFFFL)
+                            << shift;
+            resultLow |=
+                    (Float.floatToIntBits(
+                                            Float.intBitsToFloat(
+                                                            (int)
+                                                                    ((val1Low >> shift)
+                                                                            & 0xFFFFFFFFL))
+                                                    * Float.intBitsToFloat(
+                                                            (int)
+                                                                    ((val2Low >> shift)
+                                                                            & 0xFFFFFFFFL)))
+                                    & 0xFFFFFFFFL)
+                            << shift;
+        }
```

Comment:
I guess this is spotless applying some weird formatting, can you maybe assign some values to variables to avoid this weird spacing?

Reply:
Yeah, this the downside of an automatic formatter

Reply:
I'm improving a little, I'm not too worried, as in this case those implementations are just "stubbed" waiting to be fully replaced with the usage of Vector API.